src/pages/SignUp/SignUp.js

```diff
@@ -1,9 +1,341 @@
 import React, { Component } from 'react';
+import Footer from '../../components/Footer/Footer';
 import './SignUp.scss';
 
 class SignUp extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      id: '',
+      pw: '',
+      userRulePlusClassName: 'userRulePlusOff',
+      userInfoPlusClassName: 'userInfoPlusOff',
+      checked: false,
+    };
+  }
+
+  headleChange = e => {
+    this.setState({ checked: e.target.checked });
+  };
+
+  goToNext = () => {
+    this.props.history.push('/SignUp2');
+  };
+
+  handleUserRulePlus = () => {
+    if (this.state.userRulePlusClassName === 'userRulePlusOff') {
+      this.setState({ userRulePlusClassName: 'userRulePlusOn' });
+      return;
+    } else {
+      this.setState({ userRulePlusClassName: 'userRulePlusOff' });
+      return;
+    }
+  };
```

Comment:
- 위에서 말씀드린대로 boolean 값으로 바꾸시면 아래와 같은 형태로 사용할 수 있습니다.
- 결국 이 로직은 on / off를 반대 값으로 토글 해주는 것이기 때문에
```suggestion
  toggleUserRulePlus = () => {
    const { userRulePlusClassName } = this.state
    
    this.setState({ userRulePlusClassName: !userRulePlusClassName })
  };
```

Reply:
감사합니다 !